### PR TITLE
Chapter1/homework/4

### PR DESCRIPTION
**Pull request by jasonlhoward**
Hello.  This is homework 4.  Thanks.

### CODE REVIEW

File: Src/BootCamp.Chapter/Lesson4.cs [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+            return "-";
+        }
+
+        public static float PromptFloat(string message)
+        {
+            Console.Write(message);
+            string input = Console.ReadLine();
+
+            bool isFloat = float.TryParse(input, out float number);
+            if (isFloat) return number;
```
Comment (Almantask) [inline Src/BootCamp.Chapter/Lesson4.cs]: Codacy Issue found: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4927511)

**Comment by Almantask** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 6
- Added 9
           

Complexity increasing per file
==============================
- Src/BootCamp.Chapter/Lesson4.cs  6
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4927511&bid=16301668)

File: Src/BootCamp.Chapter/Lesson4.cs [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+        {            
+            string firstName = PromptString(FirstNameMessage);
+            string lastName = PromptString(LastNameMessage);
+            int age = PromptInt(AgeMessage);
+            float weight = PromptFloat(WeightMessage);
+            float height = PromptFloat(HeightMessage);
+            float bmi = CalculateBmi(weight, height);
+
+            string response = "{0} {1} is {2} years old, he has a BMI of {3} with a weight of {4} kg and a height of {5} meters.";
+            if (bmi > 0) Console.WriteLine(response, firstName, lastName, age, bmi, weight, height);
```
Comment (Almantask) [inline Src/BootCamp.Chapter/Lesson4.cs]: Codacy Issue found: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4927511)

File: Src/BootCamp.Chapter/Lesson4.cs [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+        }
+
+        public static int PromptInt(string message)
+        {
+            Console.Write(message);
+            string input = Console.ReadLine();
+
+            bool isInteger = int.TryParse(input, out int number);
+            if (isInteger) return number;
```
Comment (Almantask) [inline Src/BootCamp.Chapter/Lesson4.cs]: Codacy Issue found: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4927511)

File: Src/BootCamp.Chapter/Lesson4.cs [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+            return (gotInput) ? -1 : 0;
+        }
+
+        public static string PromptString(string message)
+        {
+            Console.Write(message);
+            string input = Console.ReadLine();
+            
+            bool gotInput = !(string.IsNullOrEmpty(input) || string.IsNullOrWhiteSpace(input));
+            if (gotInput) return input;
```
Comment (Almantask) [inline Src/BootCamp.Chapter/Lesson4.cs]: Codacy Issue found: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4927511)

File: Src/BootCamp.Chapter/Lesson4.cs [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+            Console.Write(message);
+            string input = Console.ReadLine();
+
+            bool isFloat = float.TryParse(input, out float number);
+            if (isFloat) return number;
+
+            string inputNotNumber = "\"" + input + "\"" + " is not a valid number!";
+            Console.WriteLine(inputNotNumber);
+
+            if (message.Equals(WeightMessage)) consoleInputWeight = input;
```
Comment (Almantask) [inline Src/BootCamp.Chapter/Lesson4.cs]: Codacy Issue found: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4927511)

File: Src/BootCamp.Chapter/Lesson4.cs [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+            return (gotInput) ? -1 : 0;            
+        }
+
+        public static float CalculateBmi(float weight, float height)
+        {
+            if (height <= 0 || weight <= 0)
+            {
+                Console.WriteLine("Failed calculating BMI. Reason:");
+                if (height <= 0) Console.WriteLine("Height cannot be less than zero, but was \"{0}\".", consoleInputHeight);
+                if (weight <= 0) Console.WriteLine("Weight cannot be less than zero, but was \"{0}\".", consoleInputWeight);
```
Comment (Almantask) [inline Src/BootCamp.Chapter/Lesson4.cs]: Codacy Issue found: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4927511)

File: Src/BootCamp.Chapter/Lesson4.cs [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+        public static void Demo()
+        {            
+            string firstName = PromptString(FirstNameMessage);
+            string lastName = PromptString(LastNameMessage);
+            int age = PromptInt(AgeMessage);
+            float weight = PromptFloat(WeightMessage);
+            float height = PromptFloat(HeightMessage);
+            float bmi = CalculateBmi(weight, height);
+
+            string response = "{0} {1} is {2} years old, he has a BMI of {3} with a weight of {4} kg and a height of {5} meters.";
```
Comment (Almantask) [inline Src/BootCamp.Chapter/Lesson4.cs]: Codacy Issue found: [Add the 'const' modifier to 'response'.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4927511)

File: Src/BootCamp.Chapter/Lesson4.cs [hunk 8]
```diff
@@ -0,0 +1,84 @@
...
+            string input = Console.ReadLine();
+
+            bool isFloat = float.TryParse(input, out float number);
+            if (isFloat) return number;
+
+            string inputNotNumber = "\"" + input + "\"" + " is not a valid number!";
+            Console.WriteLine(inputNotNumber);
+
+            if (message.Equals(WeightMessage)) consoleInputWeight = input;
+            if (message.Equals(HeightMessage)) consoleInputHeight = input;
```
Comment (Almantask) [inline Src/BootCamp.Chapter/Lesson4.cs]: Codacy Issue found: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4927511)

File: Src/BootCamp.Chapter/Lesson4.cs [hunk 9]
```diff
@@ -0,0 +1,84 @@
...
+            bool gotInput = !(string.IsNullOrEmpty(input) || string.IsNullOrWhiteSpace(input));            
+            return (gotInput) ? -1 : 0;            
+        }
+
+        public static float CalculateBmi(float weight, float height)
+        {
+            if (height <= 0 || weight <= 0)
+            {
+                Console.WriteLine("Failed calculating BMI. Reason:");
+                if (height <= 0) Console.WriteLine("Height cannot be less than zero, but was \"{0}\".", consoleInputHeight);
```
Comment (Almantask) [inline Src/BootCamp.Chapter/Lesson4.cs]: Codacy Issue found: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4927511)

**Comment by Almantask** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 6
- Added 8
           

Complexity increasing per file
==============================
- Src/BootCamp.Chapter/Lesson4.cs  6
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4927511&bid=16301668)

File: Src/BootCamp.Chapter/Lesson4.cs [hunk 10]
```diff
@@ -0,0 +1,84 @@
...
+            string input = Console.ReadLine();
+
+            bool isFloat = float.TryParse(input, out float number);
+            if (isFloat) return number;
+
+            string inputNotNumber = "\"" + input + "\"" + " is not a valid number!";
+            Console.WriteLine(inputNotNumber);
+
+            if (message.Equals(WeightMessage)) consoleInputWeight = input;
+            if (message.Equals(HeightMessage)) consoleInputHeight = input;
```
Comment (Almantask) [inline Src/BootCamp.Chapter/Lesson4.cs]: Codacy Issue found: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4927511)

File: Src/BootCamp.Chapter/Lesson4.cs [hunk 11]
```diff
@@ -0,0 +1,84 @@
...
+            return (gotInput) ? -1 : 0;
+        }
+
+        public static string PromptString(string message)
+        {
+            Console.Write(message);
+            string input = Console.ReadLine();
+            
+            bool gotInput = !(string.IsNullOrEmpty(input) || string.IsNullOrWhiteSpace(input));
+            if (gotInput) return input;
```
Comment (Almantask) [inline Src/BootCamp.Chapter/Lesson4.cs]: Codacy Issue found: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4927511)

File: Src/BootCamp.Chapter/Lesson4.cs [hunk 12]
```diff
@@ -0,0 +1,84 @@
...
+            return "-";
+        }
+
+        public static float PromptFloat(string message)
+        {
+            Console.Write(message);
+            string input = Console.ReadLine();
+
+            bool isFloat = float.TryParse(input, out float number);
+            if (isFloat) return number;
```
Comment (Almantask) [inline Src/BootCamp.Chapter/Lesson4.cs]: Codacy Issue found: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4927511)

File: Src/BootCamp.Chapter/Lesson4.cs [hunk 13]
```diff
@@ -0,0 +1,84 @@
...
+            if (bmi > 0) Console.WriteLine(response, firstName, lastName, age, bmi, weight, height);
+        }
+
+        public static int PromptInt(string message)
+        {
+            Console.Write(message);
+            string input = Console.ReadLine();
+
+            bool isInteger = int.TryParse(input, out int number);
+            if (isInteger) return number;
```
Comment (Almantask) [inline Src/BootCamp.Chapter/Lesson4.cs]: Codacy Issue found: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4927511)

File: Src/BootCamp.Chapter/Lesson4.cs [hunk 14]
```diff
@@ -0,0 +1,84 @@
...
+            return (gotInput) ? -1 : 0;            
+        }
+
+        public static float CalculateBmi(float weight, float height)
+        {
+            if (height <= 0 || weight <= 0)
+            {
+                Console.WriteLine("Failed calculating BMI. Reason:");
+                if (height <= 0) Console.WriteLine("Height cannot be less than zero, but was \"{0}\".", consoleInputHeight);
+                if (weight <= 0) Console.WriteLine("Weight cannot be less than zero, but was \"{0}\".", consoleInputWeight);
```
Comment (Almantask) [inline Src/BootCamp.Chapter/Lesson4.cs]: Codacy Issue found: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4927511)

File: Src/BootCamp.Chapter/Lesson4.cs [hunk 15]
```diff
@@ -0,0 +1,84 @@
...
+        {            
+            string firstName = PromptString(FirstNameMessage);
+            string lastName = PromptString(LastNameMessage);
+            int age = PromptInt(AgeMessage);
+            float weight = PromptFloat(WeightMessage);
+            float height = PromptFloat(HeightMessage);
+            float bmi = CalculateBmi(weight, height);
+
+            const string response = "{0} {1} is {2} years old, he has a BMI of {3} with a weight of {4} kg and a height of {5} meters.";
+            if (bmi > 0) Console.WriteLine(response, firstName, lastName, age, bmi, weight, height);
```
Comment (Almantask) [inline Src/BootCamp.Chapter/Lesson4.cs]: Codacy Issue found: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4927511)

File: Src/BootCamp.Chapter/Lesson4.cs [hunk 16]
```diff
@@ -0,0 +1,84 @@
...
+            Console.Write(message);
+            string input = Console.ReadLine();
+
+            bool isFloat = float.TryParse(input, out float number);
+            if (isFloat) return number;
+
+            string inputNotNumber = "\"" + input + "\"" + " is not a valid number!";
+            Console.WriteLine(inputNotNumber);
+
+            if (message.Equals(WeightMessage)) consoleInputWeight = input;
```
Comment (Almantask) [inline Src/BootCamp.Chapter/Lesson4.cs]: Codacy Issue found: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4927511)

File: Src/BootCamp.Chapter/Lesson4.cs [hunk 17]
```diff
@@ -0,0 +1,84 @@
...
+            bool gotInput = !(string.IsNullOrEmpty(input) || string.IsNullOrWhiteSpace(input));            
+            return (gotInput) ? -1 : 0;            
+        }
+
+        public static float CalculateBmi(float weight, float height)
+        {
+            if (height <= 0 || weight <= 0)
+            {
+                Console.WriteLine("Failed calculating BMI. Reason:");
+                if (height <= 0) Console.WriteLine("Height cannot be less than zero, but was \"{0}\".", consoleInputHeight);
```
Comment (Almantask) [inline Src/BootCamp.Chapter/Lesson4.cs]: Codacy Issue found: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4927511)

**Comment by Almantask** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 6
- Added 8
           

Complexity increasing per file
==============================
- Src/BootCamp.Chapter/Lesson4.cs  6
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4927511&bid=16301668)

File: Src/BootCamp.Chapter/Lesson4.cs [hunk 18]
```diff
@@ -0,0 +1,84 @@
...
```
Comment (Almantask) [inline Src/BootCamp.Chapter/Lesson4.cs]: Codacy Issue found: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4927511)

File: Src/BootCamp.Chapter/Lesson4.cs [hunk 19]
```diff
@@ -0,0 +1,84 @@
...
+            if (bmi > 0) Console.WriteLine(Response, firstName, lastName, age, bmi, weight, height);
+        }
+
+        public static int PromptInt(string message)
+        {
+            Console.Write(message);
+            string input = Console.ReadLine();
+
+            bool isInteger = int.TryParse(input, out int number);
+            if (isInteger) return number;
```
Comment (Almantask) [inline Src/BootCamp.Chapter/Lesson4.cs]: Codacy Issue found: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4927511)

File: Src/BootCamp.Chapter/Lesson4.cs [hunk 20]
```diff
@@ -0,0 +1,84 @@
...
+            Console.Write(message);
+            string input = Console.ReadLine();
+
+            bool isFloat = float.TryParse(input, out float number);
+            if (isFloat) return number;
+
+            string inputNotNumber = "\"" + input + "\"" + " is not a valid number!";
+            Console.WriteLine(inputNotNumber);
+
+            if (message.Equals(WeightMessage)) consoleInputWeight = input;
```
Comment (Almantask) [inline Src/BootCamp.Chapter/Lesson4.cs]: Codacy Issue found: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4927511)

File: Src/BootCamp.Chapter/Lesson4.cs [hunk 21]
```diff
@@ -0,0 +1,84 @@
...
+        {            
+            string firstName = PromptString(FirstNameMessage);
+            string lastName = PromptString(LastNameMessage);
+            int age = PromptInt(AgeMessage);
+            float weight = PromptFloat(WeightMessage);
+            float height = PromptFloat(HeightMessage);
+            float bmi = CalculateBmi(weight, height);
+
+            const string Response = "{0} {1} is {2} years old, he has a BMI of {3} with a weight of {4} kg and a height of {5} meters.";
+            if (bmi > 0) Console.WriteLine(Response, firstName, lastName, age, bmi, weight, height);
```
Comment (Almantask) [inline Src/BootCamp.Chapter/Lesson4.cs]: Codacy Issue found: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4927511)

File: Src/BootCamp.Chapter/Lesson4.cs [hunk 22]
```diff
@@ -0,0 +1,84 @@
...
+            return "-";
+        }
+
+        public static float PromptFloat(string message)
+        {
+            Console.Write(message);
+            string input = Console.ReadLine();
+
+            bool isFloat = float.TryParse(input, out float number);
+            if (isFloat) return number;
```
Comment (Almantask) [inline Src/BootCamp.Chapter/Lesson4.cs]: Codacy Issue found: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4927511)

File: Src/BootCamp.Chapter/Lesson4.cs [hunk 23]
```diff
@@ -0,0 +1,84 @@
...
+            return (gotInput) ? -1 : 0;            
+        }
+
+        public static float CalculateBmi(float weight, float height)
+        {
+            if (height <= 0 || weight <= 0)
+            {
+                Console.WriteLine("Failed calculating BMI. Reason:");
+                if (height <= 0) Console.WriteLine("Height cannot be less than zero, but was \"{0}\".", consoleInputHeight);
+                if (weight <= 0) Console.WriteLine("Weight cannot be less than zero, but was \"{0}\".", consoleInputWeight);
```
Comment (Almantask) [inline Src/BootCamp.Chapter/Lesson4.cs]: Codacy Issue found: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4927511)

File: Src/BootCamp.Chapter/Lesson4.cs [hunk 24]
```diff
@@ -0,0 +1,84 @@
...
+            return (gotInput) ? -1 : 0;
+        }
+
+        public static string PromptString(string message)
+        {
+            Console.Write(message);
+            string input = Console.ReadLine();
+            
+            bool gotInput = !(string.IsNullOrEmpty(input) || string.IsNullOrWhiteSpace(input));
+            if (gotInput) return input;
```
Comment (Almantask) [inline Src/BootCamp.Chapter/Lesson4.cs]: Codacy Issue found: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4927511)

File: Src/BootCamp.Chapter/Lesson4.cs [hunk 25]
```diff
@@ -0,0 +1,84 @@
...
+            string input = Console.ReadLine();
+
+            bool isFloat = float.TryParse(input, out float number);
+            if (isFloat) return number;
+
+            string inputNotNumber = "\"" + input + "\"" + " is not a valid number!";
+            Console.WriteLine(inputNotNumber);
+
+            if (message.Equals(WeightMessage)) consoleInputWeight = input;
+            if (message.Equals(HeightMessage)) consoleInputHeight = input;
```
Comment (Almantask) [inline Src/BootCamp.Chapter/Lesson4.cs]: Codacy Issue found: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4927511)

File: Src/BootCamp.Chapter/Lesson4.cs (hunk 21 above)
Comment (ellmaster) [inline Src/BootCamp.Chapter/Lesson4.cs]: I like your use of response here.
However I find it irritating that you wrote a local constant with an uppercase letter. Was this suggested by Codacy?

File: Src/BootCamp.Chapter/Lesson4.cs [hunk 26]
```diff
@@ -0,0 +1,84 @@
...
+
+        public static int PromptInt(string message)
+        {
+            Console.Write(message);
+            string input = Console.ReadLine();
+
+            bool isInteger = int.TryParse(input, out int number);
+            if (isInteger) return number;
+
+            bool gotInput = !(string.IsNullOrEmpty(input) || string.IsNullOrWhiteSpace(input));
```
Comment (ellmaster) [inline Src/BootCamp.Chapter/Lesson4.cs]: My personal preference would probably be to see this inside an extra function to make it more readable and understandable.

File: Src/BootCamp.Chapter/Lesson4.cs [hunk 27]
```diff
@@ -0,0 +1,84 @@
...
+            Console.WriteLine("\"" + input + "\"" + " is not a valid number!");
+            return (gotInput) ? -1 : 0;
+        }
+
+        public static string PromptString(string message)
+        {
+            Console.Write(message);
+            string input = Console.ReadLine();
+            
+            bool gotInput = !(string.IsNullOrEmpty(input) || string.IsNullOrWhiteSpace(input));
```
Comment (ellmaster) [inline Src/BootCamp.Chapter/Lesson4.cs]: Especially if you use it more than once.

File: Src/BootCamp.Chapter/Lesson4.cs (hunk 21 above)
Comment (jasonlhoward) [reply Src/BootCamp.Chapter/Lesson4.cs]: No. That was me. :) I will fix.

**Comment by Almantask** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 5
           

Complexity increasing per file
==============================
- Src/BootCamp.Chapter/Lesson4.cs  6
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4927511&bid=16301668)

File: Src/BootCamp.Chapter/Lesson4.cs (hunk 26 above)
Comment (jasonlhoward) [reply Src/BootCamp.Chapter/Lesson4.cs]: Will fix.

File: Src/BootCamp.Chapter/Lesson4.cs (hunk 27 above)
Comment (jasonlhoward) [reply Src/BootCamp.Chapter/Lesson4.cs]: Will fix.

**Comment by Almantask** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 5
           

Complexity increasing per file
==============================
- Src/BootCamp.Chapter/Lesson4.cs  5
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4927511&bid=16301668)

**Pull request closed without merging**